smdebug/core/hook.py

```diff
@@ -48,13 +49,29 @@
 
 
 class ScalarCache(object):
-    def __init__(self, scalar_name, scalar_val, mode, sm_metric, write_tb, write_event):
+    def __init__(
```

Comment:
@vandanavk might be able to answer this question correctly.  The ScalarCache object is not part of public facing API.  It is a cache created by save_scalar() method. The external users are not expected to create this object.  IMO this could be the reason.